pkg/messaging/mqtt/pubsub.go

```diff
@@ -96,6 +99,8 @@ func (ps pubsub) Subscribe(id, topic string, handler messaging.MessageHandler) e
 					client: client,
 					topics: []string{topic},
 				}
+
+				ps.subscriptions[id] = s
 			}
 		}
 		s.topics = append(s.topics, topic)
```

Comment:
Since `pubsub` holds a map of `subscription` objects and not pointers to them, this won't work. It is changing topics of the copy of subscription and will never reflect in the `subscriptions` map. Please double-check this implementation.

Reply:
I have made changes to this, and checked by printing the contents of the map, and running test cases of subscribing and unsubscribing, and match the subscription map with the expected value.